File: ios/Classes/FlutterMySplashScreenPlugin.m
#import "FlutterMySplashScreenPlugin.h"
 
//static FlutterMySplashScreenPlugin *_mysPluginInstance = nil;

@implementation FlutterMySplashScreenPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
   FlutterMethodChannel* channel = [FlutterMethodChannel
       methodChannelWithName:@"flutter_my_splash_screen"
             binaryMessenger:[registrar messenger]];
   FlutterMySplashScreenPlugin* instance = [[FlutterMySplashScreenPlugin alloc] init];
   [registrar addMethodCallDelegate:instance channel:channel];
//   _mysPluginInstance = instance;
        [[[UIAlertView alloc] initWithTitle:@"注册channel" message:@"Plugin" delegate:nil cancelButtonTitle:nil otherButtonTitles:@"确定", nil] show];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"show" isEqualToString:call.method]) {
      [[NSNotificationCenter defaultCenter] postNotificationName:@"flutter_my_splash_screen_show" object:nil];
  } else if ([@"hide" isEqualToString:call.method]) {
    [[[UIAlertView alloc] initWithTitle:@"接收到隐藏通知" message:@"Plugin" delegate:nil cancelButtonTitle:nil otherButtonTitles:@"确定", nil] show];
     [[NSNotificationCenter defaultCenter] postNotificationName:@"flutter_my_splash_screen_hide" object:nil];
  } else {
    result(FlutterMethodNotImplemented);
  }
}
@end
 

File: android/src/main/java/com/example/flutter_my_splash_screen/FlutterMySplashScreenPlugin.java
package com.example.flutter_my_splash_screen;
import android.app.Activity;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;
public class FlutterMySplashScreenPlugin implements MethodCallHandler {
    private final Activity activity;

    private FlutterMySplashScreenPlugin(Activity activity) {
        this.activity = activity;
    }

    public static void registerWith(Registrar registrar) {
        final MethodChannel channel = new MethodChannel(registrar.messenger(), "flutter_my_splash_screen");
        channel.setMethodCallHandler(new FlutterMySplashScreenPlugin(registrar.activity()));
    }

    @Override
    public void onMethodCall(MethodCall methodCall, Result result) {
        switch (methodCall.method) {
            case "show":
                show();
                break;
            case "hide":
                hide();
                break;
            default:
                result.notImplemented();
        }
    }

    /**
     * 打开启动屏
     */
    private void show() {
        SplashScreen.show(activity);
    }

    /**
     * 关闭启动屏
     */
    private void hide() {
        SplashScreen.hide(activity);
    }


}
